doc/introduction/measurements.rst

```diff
@@ -300,6 +300,44 @@ The decorator syntax applies equally well:
     def qnode(x, y):
         (...)
 
+The one-shot transform
+**********************
+
+Devices supporting mid-circuit measurements (defined using
+:func:`~.pennylane.measure`) and conditional operations (defined using
+:func:`~.pennylane.cond`) natively can estimate dynamic circuits by executing
+them one shot at a time. This is the default behaviour of a `~.pennylane.QNode` that has a
+device supporting mid-circuit measurements, as well as any `~.pennylane.QNode` with the
+:func:`~.pennylane.dynamic_one_shot` quantum function transform.
+As the name suggests, this transform only works for a `~.pennylane.QNode` executing with finite shots
+and it will raise an error if the device does not support mid-circuit measurements
+natively.
+The :func:`~.pennylane.defer_measurements` transform therefore remains the default for
+analytic calculations.
+
+The :func:`~.pennylane.dynamic_one_shot` transform is usually advantageous compared
+with the :func:`~.pennylane.defer_measurements` transform in the
+large-number-of-mid-circuit-measurements and small-number-of-shots limit.  This is because, unlike the
+deferred measurement principle, the method does not need an additional wire for every
+mid-circuit measurement present in the circuit. Otherwise, one generally gets
+equivalent results, so you may try both in an attempt to improve performance without
+worrying further about accuracy.
+
+The transform can be applied to a QNode as follows:
+
+.. code-block:: python
+
+    @qml.dynamic_one_shot
+    @qml.qnode(dev)
+    def my_quantum_function(x, y):
+        (...)
+
+.. warning::
+
+    Dynamic circuits executed with shots should be differentiated with the finite-difference method.
+    If the ``defer_measurements`` transform is used in analytic mode, ``backprop`` is also a viable
+    option.
```

Comment:
Sorry @vincentmr, in retrospect would you mind adding "should be differentiated with the *parameter-shift or* finite-difference methods."

We can also link to https://docs.pennylane.ai/en/stable/introduction/interfaces.html#gradients

Reply:
Should we instead say `finite-difference or backprop when post-selection is not None on some MCM`?

Reply:
Let's talk offline, I'm a little unsure what to do here :thinking: Probably lean towards removing this message and just having clear warnings in the gradient methods.